### DISCUSSION TITLE
Kiali Service Entries

### DISCUSSION THREAD

**Discussion by dimitarshenkov** [kiali/kiali#5301] [General]
We have same Service Entries for the Vault and Redis but Kiali shows them like separated points  , is it normal , can it be fixed ?
![image](https://user-images.githubusercontent.com/99493924/178716226-60e13e7b-8548-4e5a-af95-fd79d37ccdc8.png)

**Comment by israel-hdez** [comment]
Hello @dimitarshenkov .

Sorry for the delay. The contributor who is more familiar with this feature is currently on vacations.

To answer your question, yes, it is the normal behavior.
In most cases, the graph reflects the metrics that Istio records. In those metrics, Istio records the called hosts rather than the underlying ServiceEntry.

**Comment by jshaughn** [comment]
Thanks @israel-hdez , that is correct.  Note that a Service Entry is defined in a particular namespace but can be exported to many/all.  Kiali treats the service entry independently for each namespace for which it is defined/exported.

**Comment by dimitarshenkov** [comment]
@israel-hdez  we have one service  entire resource in k8s , we would like to see it like one point in Kiali not like separate for every service which uses it , so its not possible ?

**Reply by jshaughn** [reply]
As I mentioned above, a single service-entry definition can result in multiple-service entry nodes, one per namespace to which it is exported and used.   There is not currently an option to consolidate those service-entry nodes into a single node.  I'm not sure which namespace would be appropriate, I guess you're suggesting it would appear in the defining namespace only?

**Reply by CSimpiFoN** [reply]
Hello, it is more like why it handles 1 service entry like there were multiple different ones with the same name. It does not work like that if multiple services call the same other service in the mesh.
Let's say if in the example above, banners-api and jackpots-api called games-api-v2, that would not show up twice.
I can imagine that the problem actually is in the ServiceEntry implementation of Istio itself.

**Reply by dimitarshenkov** [reply]
service entrie namespace is default here yes
![image](https://user-images.githubusercontent.com/99493924/182175724-b2d37138-42ad-451a-b4b4-e3d72570b183.png)

**Reply by jshaughn** [reply]
Right, from the Istio docs, _The ‘exportTo’ field allows for control over the visibility of a service declaration to other namespaces in the mesh. By default, a service is exported to all namespaces_.  Kiali visualizes this as if each namespace has its own copy of the service-entry, which is fairly accurate from the perspective of how the traffic flows.  @CSimpiFoN says, _in the example above, banners-api and jackpots-api called games-api-v2, that would not show up twice._, which is correct **assuming** `games-api-v2` was requested in the `games-api-v2` namespace, where it lives.

Kiali could have chosen to visualize via a single service-entry node, likely in the service-entry's definition namespace, but that didn't seem like the best logical representation of the traffic.  Having said that, it's probably not an impossible option to implement, if someone was willing to add that code.

**Reply by dimitarshenkov** [reply]
Ok thanks a lot for clarification
